### PR TITLE
Logging work

### PR DESCRIPTION
**Pull request by mponomar**
More logging work:

- Fingerprints for write statements
- Client can set context, recorded in event log
- More information in event logs: tables, plan, errors
- pmux patch to keep track of currently running dbs
- Beginning of consumer program for event logs

### CODE REVIEW

**Comment by jmleddy** [comment]
Looks good

**Comment by mponomar** [comment]
Thanks, waiting for roborivers.

**Comment by jmleddy** [comment]
Where is roborivers?

**Comment by mponomar** [comment]
Roborivers is running, and found lots of failures.

**Comment by jmleddy** [comment]
Oh no

**Comment by riverszhang89** [comment]
```
Core was generated by `/home/admin/branches/master/comdb2 perfectckp31611 -pidfile /home/admin/branche'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000000007ab510 in fingerprintExpr (db=0x8cd4d98, c=0x7fcebd9b48f0, p=0x1) at sqlite/select.c:88
88          if (p == NULL || p->visited)
(gdb) p p
$1 = (Expr *) 0x1
```

**Comment by riverszhang89** [comment]
Another one:
```
#0  0x00000000007ab510 in fingerprintExpr (db=0x6395a78, c=0x7f777bb98290, p=0x3) at sqlite/select.c:88
88          if (p == NULL || p->visited)
```

p seems trashed.

**Comment by mponomar** [comment]
Thanks (robo)?rivers! I think I have a patch for this.  Need to call the fingerprint code after its parsed, but before the corresponding planner routine is called.

File: db/comdb2.c [hunk 1]
```diff
@@ -631,7 +631,7 @@ extern int gbl_parallel_memptrickle;
 int gbl_verify_dbreg = 0;
 extern int gbl_checkpoint_paranoid_verify;
 
-int gbl_forbid_ulonglong = 1;
+int gbl_forbid_ulonglong = 0;
```
Comment (adizaimi) [inline db/comdb2.c]: wow, we allow ulonglong back?

**Pull request merged**